File: src/api/subject/Dto.ts
export interface SubjectResDto {
    id: string,

    name: string,

    description: string,

    auth: boolean,

    created_at: string,

    updated_at: string,
}

export interface SubjectCreateDto {

    name: string,

    description: string,

    belong_category: string,
}

export interface SubjectUpdateDto {

    id: string,

    name?: string,

    description?: string,

    auth?: string,
}

export interface QuerySubjectDto {
    id?: string;

    name?: string;

    belong_category?: string;

    order_by?: string;

    limit?: number;

    start?: number;
}

File: src-tauri/src/modules/subject/application/query/query_subject/dto.rs
use serde::{Deserialize, Serialize};


#[derive(Serialize, Deserialize)]
pub struct QuerySubjectDto { 
    pub id: Option<String>,

    pub name: Option<String>,

    pub belong_category: Option<String>, 

    pub order_by: Option<String>,
    
    pub limit: Option<i64>,

    pub start: Option<i64>,
}
